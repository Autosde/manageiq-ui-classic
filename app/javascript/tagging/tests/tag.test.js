import React from 'react';
import Tag from '../components/tag';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const tagCategory = 'animal';
const tagValue = 'duck';
function onDelete(x) {
  return x;
}

describe('Tag Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(<Tag tagCategory={tagCategory} tagValue={tagValue} onTagDeleteClick={onDelete} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call methods', () => {
    const onDelete = jest.fn();
    const wrapper = shallow(<Tag tagCategory={tagCategory} tagValue={tagValue} onTagDeleteClick={onDelete} />);
    wrapper.instance().handleClick('Knedlik');
    expect(onDelete.mock.calls.length).toEqual(1);
  });
});
