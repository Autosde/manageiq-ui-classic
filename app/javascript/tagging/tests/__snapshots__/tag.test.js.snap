// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component match snapshot 1`] = `
<li>
  <span
    className="label label-info"
  >
    animal
    : 
    duck
    <a
      href="#"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
        className="pficon pficon-close"
      />
      <span
        className="sr-only"
      >
        Remove
      </span>
    </a>
  </span>
</li>
`;
